azure-core/azure/core/pipeline/__init__.py

```diff
@@ -47,7 +47,7 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 class PipelineContext(dict):
 
-    def __init__(self, transport, **kwargs):
+    def __init__(self, transport, **kwargs): #pylint: disable=super-init-not-called
```

Comment:
This warning should not be disabled - instead we actually have a bug. We need to add the super init call here.

Reply:
calling super init is causing half of the tests to fail.; still looking into it.
out of curiosity, is there a reason [UserDict](https://docs.python.org/2/library/userdict.html#module-UserDict) or [MutableMapping](https://docs.python.org/3/library/collections.abc.html#collections.abc.MutableMapping) is not used?